### ISSUE TITLE
Pylint * not working on Windows

### ISSUE THREAD

**Issue by AndreMPCosta** [PyCQA/pylint#8009]
### Bug description

If I run pylint * on my Windows machine I get:

```
************* Module *

*:1:0: F0001: No module named * (fatal)
```

If I run this on my Mac machine it works as expected.

```
Your code has ben rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Command used

```shell
pylint *
```

### Pylint output

```shell
*:1:0: F0001: No module named * (fatal)
```

### Expected behavior

Your code has ben rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Pylint version

```shell
pylint 2.15.9
astroid 2.12.13
Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 11

**Comment by Pierre-Sassoulas** [comment]
I think  ``*`` on Windows means the literal character ``*`` unlike in unix where it's going to expand to all the files in the directory. You probably want to use ``pylint . --recursive=y`` instead.

**Comment by AndreMPCosta** [comment]
It was working fine with `*` but I'll use the `pylint . --recursive=y` instead.

**Comment by Pierre-Sassoulas** [comment]
Was it running on windows before ? (Did you upgrade ?)

**Comment by AndreMPCosta** [comment]
Yep it was running fine on Windows (the `pylint *`), but I've downgraded pylint and still had the same issue, so the problem should be on Windows and not on pylint.

**Comment by AndreMPCosta** [comment]
I was using Pycharm terminal, I didn't change from it, so I can't pinpoint what triggered this.
